pkg/chunked/storage_linux.go

```diff
@@ -582,7 +601,10 @@ func (o *originFile) OpenFile() (io.ReadCloser, error) {
 }
 
 // setFileAttrs sets the file attributes for file given metadata
-func setFileAttrs(dirfd int, file *os.File, mode os.FileMode, metadata *internal.FileMetadata, options *archive.TarOptions, usePath bool) error {
+func setFileAttrs(dirfd int, file *os.File, mode os.FileMode, metadata *fileMetadata, options *archive.TarOptions, usePath bool) error {
```

Comment:
(Unrelated: if I’m not mistaken, `mode` is always `os.FileMode(metadata.Mode)`, so the parameter could be eliminated throughout the call stack. Perhaps best to do in a separate PR.)